pulse2percept/implants/base.py

```diff
@@ -347,18 +347,20 @@ class ElectrodeGrid(ElectrodeArray):
     * Allow user-specified electrode types (see :issue:`122`).
 
     """
-
-    def __init__(self, shape, x=0, y=0, z=0, rot=0, r=10, spacing=None,
-                 names=('A', '1')):
+    
+    def __init__(self, shape, x=0, y=0, z=0, rot=0, r = 10, spacing=None,
```

Comment:
You are on the right track. Because different electrode objects have different number of parameters, it would be better change the function signature to:

```
def __init__(self, shape, x=0, y=0, z=0, rot=0, spacing=None,
                  electrode_type='DiskElectrode', electrode_kwargs=None):
```

where `electrode_kwargs` is a dictionary with optional keyword arguments for `electrode_type`.

So the user could do:
- `ElectrodeGrid((4, 4), electrode_type='PointSource')
- `ElectrodeGrid((4, 4), electrode_type='DiskElectrode', electrode_kwargs={'r': 10})`

This way, we don't need to change the function signature when we add more electrode types in the future.